gammapy/data/data_store.py

```diff
@@ -284,15 +289,29 @@ def obs(self, obs_id, required_irf="full-enclosure"):
 
         kwargs = {"obs_id": int(obs_id)}
 
-        if required_irf == "full-enclosure":
-            hdus = ["events", "gti", "aeff", "edisp", "psf", "bkg"]
-        elif required_irf == "point-like":
-            hdus = ["events", "gti", "aeff", "edisp"]
-        else:
-            hdus = ["events", "gti"] + required_irf
+        # check for the "short forms"
+        if isinstance(required_irf, str):
+            required_irf = REQUIRED_IRFS[required_irf]
 
-        for hdu in hdus:
-            kwargs[hdu] = self.hdu_table.hdu_location(obs_id=obs_id, hdu_type=hdu)
+        if not set(required_irf).issubset(ALL_IRFS):
+            difference = set(required_irf).difference(ALL_IRFS)
+            raise ValueError(
+                f"{difference} is not a valid hdu key. Choose from: {ALL_IRFS}"
+            )
+
+        required_hdus = {"event", "gti"}.union(required_irf)
+
+        for hdu in ALL_HDUS:
+            hdu_location = self.hdu_table.hdu_location(
+                obs_id=obs_id, hdu_type=hdu, warn_missing=False,
+            )
+            if hdu_location is not None:
+                kwargs[hdu] = hdu_location
+            else:
+                if hdu in required_hdus:
+                    raise MissingRequiredHDU(
+                        f"Required HDU {hdu} not found in observation {obs_id}"
+                    )
```

Comment:
Multiple hdus may be missing while the raise will catch only the first one that fails so I modified my previous suggestion